include/eve/algo/converting_iterator.hpp

```diff
@@ -0,0 +1,112 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/algo/iterator_helpers.hpp>
+
+#include <eve/function/convert.hpp>
+#include <eve/function/load.hpp>
+#include <eve/function/store.hpp>
+
+namespace eve::algo
+{
+  template <typename I, typename T>
+  struct converting_iterator : operations_with_distance
+  {
+    using cardinal = typename I::cardinal;
+    using value_type = T;
+    using wide_value_type = eve::wide<value_type, cardinal>;
+
+    converting_iterator() = default;
+    explicit converting_iterator(I base) : base(base) {}
+
+    operator converting_iterator<unaligned_t<I>, T>() const
+      requires (!unaligned_iterator<I>)
+    {
+      return converting_iterator<unaligned_t<I>, T>{base};
+    }
+
+    converting_iterator<unaligned_t<I>, T> unaligned() const { return *this; }
+    auto previous_partially_aligned() const
+    {
+      return converting_iterator<partially_aligned_t<I>, T>{base.previous_partially_aligned()};
+    }
+
+    template <typename _Cardinal>
+    auto cardinal_cast(_Cardinal N) const
+    {
+      auto new_base = base.cardinal_cast(N);
+      return converting_iterator<decltype(new_base), T>{new_base};
+    }
+
+    T operator*() const { return static_cast<T>(*base); }
+
+    template <sentinel_for<I> I2>
+    friend bool operator==(converting_iterator const& x, converting_iterator<I2, T> const& y)
+    {
+      return x.base == y.base;
+    }
+
+    template <sentinel_for<I> I2>
```

Comment:
why both == and <=> ?

Reply:
I think either compiler or I don't understand these. Maybe the one from the base class get's picked up.

Basically, doesn't work otherwise.

Reply:
OK maybe we need to have a deeper look but this is fine for now